### PR TITLE
[MINOR] migrate TransactionPool (& affiliated test) from 'core' to 'eth'

### PR DESCRIPTION
**Pull request by s-matthew-english**
## PR description

Necessary pre-condition of using `SyncState` in `TransactionPool` as a filtering mechanism. 

The gotcha was adding these lines: 

```
testImplementation project(path: ':config', configuration: 'testSupportArtifacts')
integrationTestImplementation project(path: ':config', configuration: 'testSupportArtifacts')
```

to `ethereum/eth/build.gradle`, since `TransactionPoolTest` has `ExecutionContextTestFixture` which requires `GenesisConfigFile`, or anyway one of those files in `tech.pegasys.pantheon.config`.

### CODE REVIEW

File: ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/PendingTransactions.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public boolean addRemoteTransaction(final Transaction transaction) {
     return addTransaction;
   }
 
-  boolean addLocalTransaction(final Transaction transaction) {
+  public boolean addLocalTransaction(final Transaction transaction) {
```
Comment (mbaxter) [inline ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/PendingTransactions.java]: You should move all of the `TransactionPool` related classes to `eth`.  I believe that includes: `TransactionPool`, `PendingTransactions`, `PendingTransactionListener`, `PendingTransactionDroppedListener`.

File: ethereum/eth/build.gradle [hunk 2]
```diff
@@ -55,4 +55,7 @@ dependencies {
   testImplementation 'org.mockito:mockito-core'
 
   jmhImplementation project(path: ':ethereum:core', configuration: 'testSupportArtifacts')
+
+  testImplementation project(path: ':config', configuration: 'testSupportArtifacts')
```
Comment (shemnon) [inline ethereum/eth/build.gradle]: This like should be alphabetized with the rest of the testImplementation dependenceies.  Between line 45 and 46

Comment (s-matthew-english) [reply ethereum/eth/build.gradle]: Nice catch -- thank you.

File: ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/PendingTransactions.java (hunk 1 above)
Comment (s-matthew-english) [reply ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/PendingTransactions.java]: Yeah -- I was initially wondering about whether or not to do that -- I've updated the branch accordingly

**Pull request merged**